src/tracing_integration.rs

```diff
@@ -0,0 +1,115 @@
+use crate::error::RadosResult;
+use crate::rados::*;
+
+use libc::*;
+use std::ffi::CStr;
+use tracing::Level;
+
+/// For tracing integration to function properly, logging (tracing handler) must be enabled before
+/// connecting to ceph: if tracing is not enabled, we don't set a callback, which avoids the
+/// corresponding overhead.
+pub(crate) fn enable_tracing_integration(rados: rados_t) -> RadosResult<()> {
+    let (level_str, opt_log_callback): (&[u8], rados_log_callback_t) = {
+        if event_enabled!(target: "librados", Level::ERROR) {
```

Comment:
Hi, I assume you're writing this way to minimize the runtime cost of additional checks. However, the enable API itself is usually not a heavy call. How about expanding them into one place within the loop?

for example:

```rust
type RadosLogCallback = rados_log_callback_t;

fn librados_log_level() -> (&'static [u8], Option<RadosLogCallback>) {
    const CANDIDATES: &[(Level, &[u8])] = &[
        (Level::DEBUG, b"debug\0"),
        (Level::INFO,  b"info\0"),
        (Level::WARN,  b"warn\0"),
        (Level::ERROR, b"error\0"),
    ];

    for &(lvl, cstr) in CANDIDATES {
        if event_enabled!(target: "librados", lvl) {
            return (cstr, Some(log_callback));
        }
    }

    (b"\0", None)
}
```

Reply:
I think that is not possible because these macros require const level, like below.